azure-go-aks/main.go

```diff
@@ -135,6 +144,81 @@ func main() {
 
 		// Export the raw kube config.
 		ctx.Export("kubeconfig", cluster.KubeConfigRaw)
+
+		k8sProvider, err := providers.NewProvider(ctx, "k8sprovider", &providers.ProviderArgs{
+			Kubeconfig: cluster.KubeConfigRaw,
+		}, pulumi.DependsOn([]pulumi.Resource{cluster}))
+		if err != nil {
+			return err
+		}
+
+		ns, err := corev1.NewNamespace(ctx, "app-ns", &corev1.NamespaceArgs{
+			Metadata: &metav1.ObjectMetaArgs{
+				Name: pulumi.String("example-ns"),
+			},
+		}, pulumi.Provider(k8sProvider))
+		if err != nil {
+			return err
+		}
+
+		appLabels := pulumi.StringMap{
+			"app": pulumi.String("iac-workshop"),
+		}
+		_, err = appsv1.NewDeployment(ctx, "app-dep", &appsv1.DeploymentArgs{
+			Metadata: &metav1.ObjectMetaArgs{
+				Namespace: ns.Metadata.Elem().Name(),
+			},
+			Spec: appsv1.DeploymentSpecArgs{
+				Selector: &metav1.LabelSelectorArgs{
+					MatchLabels: appLabels,
+				},
+				Replicas: pulumi.Int(3),
+				Template: &corev1.PodTemplateSpecArgs{
+					Metadata: &metav1.ObjectMetaArgs{
+						Labels: appLabels,
+					},
+					Spec: &corev1.PodSpecArgs{
+						Containers: corev1.ContainerArray{
+							corev1.ContainerArgs{
+								Name:  pulumi.String("iac-workshop"),
+								Image: pulumi.String("jocatalin/kubernetes-bootcamp:v2"),
+							}},
+					},
+				},
+			},
+		}, pulumi.Provider(k8sProvider))
+		if err != nil {
+			return err
+		}
+
+		service, err := corev1.NewService(ctx, "app-service", &corev1.ServiceArgs{
+			Metadata: &metav1.ObjectMetaArgs{
+				Namespace: ns.Metadata.Elem().Name(),
+				Labels:    appLabels,
+			},
+			Spec: &corev1.ServiceSpecArgs{
+				Ports: corev1.ServicePortArray{
+					corev1.ServicePortArgs{
+						Port:       pulumi.Int(80),
+						TargetPort: pulumi.Int(8080),
+					},
+				},
+				Selector: appLabels,
+				Type:     pulumi.String("LoadBalancer"),
+			},
+		}, pulumi.Provider(k8sProvider))
+		if err != nil {
+			return err
+		}
+
+		ctx.Export("url", service.Status.ApplyT(func(status *corev1.ServiceStatus) *string {
+			ingress := status.LoadBalancer.Ingress[0]
```

Comment:
I'm not sure if this will work as AKS LB is in preview last time I checked. This requires we create a static ip and set it in the loadbalancer, with a particular network role assignment:

- network role to link static ip to LB: https://github.com/metral/multicloud-clusters/blob/master/aks.ts#L45-L51
- creating a static ip: https://github.com/metral/multicloud-clusters/blob/master/aks.ts#L104-L108
- using the static ip in the LB:https://github.com/metral/multicloud-apps/blob/master/app.ts#L42

Reply:
Yeah, this works, I was able to test this got a correct IP from Azure :)